File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario1/id_29T2000Test.py
# id_29_Test.java
from scenario1.id_29 import filter_by_prefix

import pytest

class id_29_Test:
    """
    Test class of id_29.py.
    It contains ten unit test cases for the :func:`scenario1.id_29.filter_by_prefix`.
    """


    def test_filter_by_prefix_1(self):
        """
        Test case 1 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "ab") == ["abc", "abcd", "abcde"]

    def test_filter_by_prefix_2(self):
        """
        Test case 2 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "abc") == ["abc", "abcd", "abcde"]

    def test_filter_by_prefix_3(self):
        """
        Test case 3 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "abcd") == ["abcd"]

    def test_filter_by_prefix_4(self):
        """
        Test case 4 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "abcde") == ["abcde"]

    def test_filter_by_prefix_5(self):
        """
        Test case 5 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "abcdef") == []

    def test_filter_by_prefix_6(self):
        """
        Test case 6 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "") == ["abc", "abcd", "abcde"]

    def test_filter_by_prefix_7(self):
        """
        Test case 7 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "a") == ["abc", "abcd", "abcde"]

    def test_filter_by_prefix_8(self):
        """
        Test case 8 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "b") == []

    def test_filter_by_prefix_9(self):
        """
        Test case 9 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "c") == []

    def test_filter_by_prefix_10(self):
        """
        Test case 10 for :func:`scenario1.id_29.filter_by_prefix`.
        """
        assert filter_by_prefix(["abc", "abcd", "abcde"], "d") == []

File: RQ2_Prompt_Elements/StarCoder_Data/HumanEvalJava_output/scenario1/id_29T2000Test.java
// FilterByPrefixTest.java
package scenario1;

import java.util.ArrayList;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link FilterByPrefix}.
* It contains ten unit test cases for the {@link FilterByPrefix#filterByPrefix(List, String)} method.
*/
class FilterByPrefixTest {

    @Test
    void testFilterByPrefix() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abc");
        assertEquals(10, result.size());
    }

    @Test
    void testFilterByPrefix2() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcd");
        assertEquals(9, result.size());
    }

    @Test
    void testFilterByPrefix3() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcde");
        assertEquals(8, result.size());
    }

    @Test
    void testFilterByPrefix4() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcdef");
        assertEquals(7, result.size());
    }

    @Test
    void testFilterByPrefix5() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcdefg");
        assertEquals(6, result.size());
    }

    @Test
    void testFilterByPrefix6() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcdefgh");
        assertEquals(5, result.size());
    }

    @Test
    void testFilterByPrefix7() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcdefghi");
        assertEquals(4, result.size());
    }

    @Test
    void testFilterByPrefix8() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcdefghij");
        assertEquals(3, result.size());
    }

    @Test
    void testFilterByPrefix9() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcdefghijk");
        assertEquals(2, result.size());
    }

    @Test
    void testFilterByPrefix10() {
        List<Object> strings = new ArrayList<Object>();
        strings.add("abc");
        strings.add("abcd");
        strings.add("abcde");
        strings.add("abcdef");
        strings.add("abcdefg");
        strings.add("abcdefgh");
        strings.add("abcdefghi");
        strings.add("abcdefghij");
        strings.add("abcdefghijk");
        strings.add("abcdefghijkl");
        List<Object> result = FilterByPrefix.filterByPrefix(strings, "abcdefghijkl");
        assertEquals(1, result.size());
    }
}
